src/shell/Session.ts

```diff
@@ -112,7 +113,32 @@ export class Session extends EmitterWithUniqueID {
 
     private deserialize(): void {
         this.directory = this.readSerialized(presentWorkingDirectoryFilePath, homeDirectory);
-        History.deserialize(this.readSerialized(historyFilePath, []));
+        let bashHistoryModifiedTime = new Date(0);
+        let bashHistory: string[] = [];
+        try {
+            bashHistoryModifiedTime = statSync(bashHistoryFilePath).mtime;
+            bashHistory = readFileSync(bashHistoryFilePath).toString().trim().split("\n").reverse();
+        } catch (error) {
+            // File was missing or something. Ignore.
+        }
+
+        let blackScreenHistoryModifiedTime = new Date(0);
+        let blackScreenHistory: string[] = [];
+        try {
+            blackScreenHistoryModifiedTime = statSync(historyFilePath).mtime;
+            blackScreenHistory = this.readSerialized(historyFilePath, []);
+        } catch (error) {
+            // File was missing or something. Ignore.
+        }
+
+        let mergedHistory: string[] = [];
+        if (blackScreenHistoryModifiedTime > bashHistoryModifiedTime) {
+            mergedHistory = [...bashHistory, ...blackScreenHistory];
+        } else {
+            mergedHistory = [...blackScreenHistory, ...bashHistory];
```

Comment:
Would it be possible to interleave the records based on time? It's not trivial to do, so I'm OK with a "no".